Bio/SeqIO/NibIO.py

```diff
@@ -80,27 +80,30 @@ def __init__(self, source):
 
         """
         super().__init__(source, mode="b", fmt="Nib")
-
-    def parse(self, handle):
-        """Start parsing the file, and return a SeqRecord generator."""
-        word = handle.read(4)
+        word = self.stream.read(4)
         if not word:
             raise ValueError("Empty file.")
         signature = word.hex()
         if signature == "3a3de96b":
-            byteorder = "little"  # little-endian
+            self.byteorder = "little"  # little-endian
         elif signature == "6be93d3a":
-            byteorder = "big"  # big-endian
+            self.byteorder = "big"  # big-endian
         else:
             raise ValueError("unexpected signature in nib header")
-        records = self.iterate(handle, byteorder)
-        return records
 
-    def iterate(self, handle, byteorder):
+    def parse(self, stream):
+        """To be removed."""
+        return
+
+    def __next__(self):
         """Iterate over the records in the nib file."""
-        number = handle.read(4)
+        stream = self.stream
+        byteorder = self.byteorder
+        number = stream.read(4)
+        if len(number) == 0:
```

Comment:
I understand "if not number:" as per the original (bar the variable name change) to be considered more Pythonic.

Reply:
I was thinking to separate `len(number) == 0` (and raising a `StopIteration`) from `0 < len(number) < 4` (and raise a `ValueError`), but since we are not considering the latter option now, I switched back `if not number`.